src/components/plan/PlanForm.jsx

```diff
@@ -88,117 +94,84 @@ const PlanForm = () => {
     };
 
     return (
-        <section className="
-                text-gray-600
-                body-font
-                relative
-        ">
-            <div className="
-                container
-                px-5
-                py-24
-                mx-auto
-            ">
-                <div className="
-                    flex
-                    flex-col
-                    text-center
-                    w-full
-                    mb-12
-                ">
-                    <h1 className="
-                        sm:text-3xl
-                        text-2xl
-                        font-medium
-                        title-font
-                        mb-4
-                        text-gray-900
-                    ">
-                        Create a new plan
-                    </h1>
-                    <p className="
-                        lg:w-2/3
-                        mx-auto
-                        leading-relaxed
-                        text-base
-                    ">
-                        Fill out the form below to create a new plan.
-                    </p>
-                </div>
-                <div className="
-                    lg:w-1/2
-                    md:w-2/3
-                    mx-auto
-                ">
-                    <div className="
-                        flex
-                        flex-wrap
-                        -m-2
-                    ">
-                        <div className="p-2 w-1/2">
-                            <Input
-                                label={"title"}
-                                type={"text"}
-                                value={title}
-                                onChange={setTitle}
-                                error={error}
-                                maxLength={30}
-                            />
+        <div>
+            {loading && (
```

Comment:
I believe that `loading && ()` conditional can be abstracted to the `Loader` component because it should always have the same behavior.

Reply:
I think the loader condition should be outside, the abstraction should keep only the object.

In my opinion the visibility toggle of the loader is a feature of the big component, not about the loader.

Waiting to your reply... 👍

Reply:
Well, in my opinion the component has to contain the elements that define its behavior. All the loader I expect to be there when in loading, and it goes when the loading end.
**"visibility toggle of the loader is a feature of the big component"**. I think that `PlanForm` and `PlanView` don't have to deal with the loaders behaviour and can be abstracted.

Reply:
I think you may be right. I think maybe I'm thinking of a `LoadView` but it's just the `Loader` in our case.... Approved ✅